broker/src/main/java/org/apache/rocketmq/broker/schedule/ScheduleMessageService.java

```diff
@@ -398,7 +398,12 @@ public void run() {
             } catch (Exception e) {
                 // XXX: warn and notify me
                 log.error("ScheduleMessageService, executeOnTimeup exception", e);
-                this.scheduleNextTimerTask(this.offset, DELAY_FOR_A_PERIOD);
+                try {
+                    // Wait for a period of time if an error has occurred.
+                    TimeUnit.MILLISECONDS.sleep(DELAY_FOR_A_PERIOD);
+                } catch (InterruptedException ex) {
+                    log.error("ScheduleMessageService, an error occurred while waiting", ex);
```

Comment:
IMO interrupted flag should be restored (`Thread.currentThread().interrupt()`) after  `InterruptedException` is thrown.